### PR TITLE
Store record keys as 64-bit integers

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

File: src/dhtnode/storage/StorageEngine.d [hunk 1]
```diff
@@ -652,7 +652,7 @@ public class StorageEngine : IStorageEngine
 
         key_ = cast(void*)tcmdbiternext(this.db, &len);
 
-        bool found = !!key_;
+        bool found = key_ !is null;
```
Comment (david-eckardt-sociomantic) [inline src/dhtnode/storage/StorageEngine.d]: Suggestion while you're at it:
```d
if (void* key_ = tcmdbiternext(this.db, &len))
{
    // ...
    return true;
}
else
    return false;
```

File: src/dhtnode/storage/StorageEngine.d [hunk 2]
```diff
@@ -210,12 +213,14 @@ public class StorageEngine : IStorageEngine
...
+        auto hash = Hash.straightToHash(key);
+
         int len;
 
         void* value_;
 
-        value_ = cast(void*)tcmdbget(this.db, key.ptr,
-            castFrom!(size_t).to!(int)(key.length), &len);
+        value_ = cast(void*)tcmdbget(this.db, &hash,
+            castFrom!(size_t).to!(int)(hash.sizeof), &len);
```
Comment (david-eckardt-sociomantic) [inline src/dhtnode/storage/StorageEngine.d]: :warning: Esoteric `!!value` below.

File: src/dhtnode/storage/StorageEngine.d [hunk 3]
```diff
@@ -647,20 +659,21 @@ public class StorageEngine : IStorageEngine
...
         int len;
+        hash_t* key_;
+        key_ = cast(hash_t*)tcmdbiternext(this.db, &len);
 
-        void* key_;
-
-        key_ = cast(void*)tcmdbiternext(this.db, &len);
-
-        bool found = !!key_;
+        bool found = key_ !is null;
```
Comment (david-eckardt-sociomantic) [inline src/dhtnode/storage/StorageEngine.d]: Suggestion while you're at it:

```d
if (void* key_ = tcmdbiternext(this.db, &len))
{
    // ...
    return true;
}
else
    return false;
```

**Comment by gavin-norman-sociomantic** [comment]
Updated.

**Pull request merged**